src/ring_theory/polynomial/hermite.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2023 Luke Mantle. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Luke Mantle
+-/
+
+import data.polynomial.derivative
+
+/-!
+# Hermite polynomials
+
+This file defines `hermite n`, the nth probabilist's Hermite polynomial.
+
+## Main definitions
+
+* `x_sub_dx`  : the operation `(x - d/dx)` used to recursively define the Hermite polynomials
+* `hermite n` : the nth probabilist's Hermite polynomial, defined as a `polynomial ℤ`, using
+                `x_sub_dx` recursively
+
+## References
+
+[Hermite Polynomials](https://en.wikipedia.org/wiki/Hermite_polynomials)
+
+-/
+
+noncomputable theory
+open polynomial
+
+section x_sub_dx
+
+variables {R : Type} [comm_ring R] (p : polynomial R)
+
+/-- the operation `(x - d/dx)` defined for `polynomial R` -/
+def x_sub_dx := X * p - p.derivative
```

Comment:
I would be inclined to remove this definition and just write it out in full when you need it to prove things about `hermite`

Reply:
I have it defined as its own function because I do use it (and an `x_sub_dx_fn` for real functions) in other files. It comes up when I show the equivalence to the definition that uses nth derivatives of the Gaussian function and the explicit formula for coefficients, and it keeps the algebra a bit neater. Would it be possible to make it a localized definition, or do you think it's best to remove it altogether?

Reply:
I would agree with Eric that this definition is not too useful. You can always use `λ p, X * p - p.derivative` if you need the operator.
There is a certain cost to adding new definitions in the form of boilerplate lemmas.